[RHCLOUD-20865] Refactor tenancy lookup orgid preference

MikelAlejoBR opened the pull request:
Since the EBS account numbers are deprecated, we want to explicitly find
the tenants by their OrgId first. We are aware that this solution
hurts the performance just a bit, but this will eventually go away.

## Links 

[[RHCLOUD-20865]](https://issues.redhat.com/browse/RHCLOUD-20865)

MikelAlejoBR committed:
refactor: prefer OrgIds when fetching / creating the tenant

Since the EBS account numbers are deprecated, we want to explicitly find
the tenants by their OrgId first. We are aware that this solution
hurts the performance just a bit, but this will eventually go away.

MikelAlejoBR committed:
refactor: log the error message when the tenant is not found / created

We were returning the error back to the client, but we were not logging
the error with all the details.

lindgrenj6 reviewed (commented):
I know this is built on #521 - but still looking :eyes: 

I think we should log when we have to fallback to the EBS acct number - that way we can at least hopefully get an idea around how many misses we're getting via kibana.

Other than that looks great!

MikelAlejoBR committed:
feature: log tenants found by their external accounts

This will help us identify which tenants are still being fetched by
their EBS account numbers.

lindgrenj6 reviewed (approved):
Looks good. I noticed we don't log not-found requests - which is probably fine but I was puzzled for a minute why I was only seeing the successful queries haha.

MikelAlejoBR committed:
feature: add tenant creation log

Helps us track which tenants weren't found neither by their org id or
ebs account numbers.

lpichler reviewed (approved):
LGTM
